app/controllers/service_provider_controller.rb

```diff
@@ -2,11 +2,27 @@ class ServiceProviderController < ApplicationController
   protect_from_forgery with: :null_session
 
   def update
-    if FeatureManagement.use_dashboard_service_providers?
-      ServiceProviderUpdater.new.run
-      SecureHeadersWhitelister.new.run
+    authorize do
+      if FeatureManagement.use_dashboard_service_providers?
+        ServiceProviderUpdater.new.run
+        SecureHeadersWhitelister.new.run
+      end
+
+      render json: { status: 'If the feature is enabled, service providers have been updated.' }
+    end
+  end
+
+  private
+
+  def authorize
+    if authorization_token == Figaro.env.dashboard_api_token
+      yield
+    else
+      render nothing: true, status: 401
     end
+  end
 
-    render json: { status: 'If the feature is enabled, service providers have been updated.' }
+  def authorization_token
+    request.headers['X-LOGIN-DASHBOARD-TOKEN']
```

Comment:
The HTTP header on the wire is of the format `X-Foo-bar`

Most languages following the CGI spec convert internally because CGI uses/abuses env vars. So the code on the server side ends up seeing `HTTP_X_FOO_BAR`. The `HTTP_` prefix is to indicate "this thing came over the wire via the HTTP exchange".